csharp/common/exception/ErrorMessage.cs

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using System;
+
+namespace com.vaticle.typedb.driver.Common.Exception
+{
+    public abstract class ErrorMessage
+    {
+        public ErrorMessage(
+            string codePrefix, int codeNumber, string messagePrefix, string messageBody)
+        {
+            _codePrefix = codePrefix;
+            _codeNumber = codeNumber.ToString("D2");
+            _messagePrefix = messagePrefix;
+            _messageBody = messageBody;
+        }
+
+        public string ToString(params object?[] errorParams)
+        {
+            var formattedBody = String.Format(_messageBody, errorParams);
+            return $"[{_codePrefix}{_codeNumber}]{_messagePrefix}: {formattedBody}";
+        }
+
+        private readonly string _codePrefix;
+        private readonly string _codeNumber;
+        private readonly string _messagePrefix;
+        private readonly string _messageBody;
+    }
+
+    namespace Error
+    {
+        public class DriverErrorMessage : ErrorMessage
+        {
+            public DriverErrorMessage(int codeNumber, string message)
+                : base(s_codePrefix, codeNumber, s_messagePrefix, message)
+            {
+            }
+
+            private static readonly string s_codePrefix = "CCL";
```

Comment:
These prefixes seem strange/mismatched.
Also let's discuss why we had to split the Driver/DriverErrorMessage briefly tomorrow

Reply:
Fixed in https://github.com/vaticle/typedb-driver/commit/204555cb2ff2a2d0b841751a2b56a2219fc66f30